repositories/src/main/scala/cromwell/pipeline/datastorage/dto/ProjectSearch.scala

```diff
@@ -0,0 +1,92 @@
+package cromwell.pipeline.datastorage.dto
+
+import cromwell.pipeline.utils.json.AdtJsonFormatter._
+import play.api.libs.functional.syntax.{ toFunctionalBuilderOps, toInvariantFunctorOps, unlift }
+import play.api.libs.json.Format.GenericFormat
+import play.api.libs.json.{ Json, __, _ }
+
+sealed trait ProjectSearchFilter
+case object All extends ProjectSearchFilter
+final case class ByName(mode: NameSearchMode, value: String) extends ProjectSearchFilter
+final case class ByConfig(mode: ContentSearchMode, value: Boolean) extends ProjectSearchFilter
+final case class ByFiles(mode: ContentSearchMode, value: Boolean) extends ProjectSearchFilter
+final case class Or(left: ProjectSearchFilter, right: ProjectSearchFilter) extends ProjectSearchFilter
+final case class And(left: ProjectSearchFilter, right: ProjectSearchFilter) extends ProjectSearchFilter
+
+object ProjectSearchFilter {
+  implicit val allFormat: OFormat[All.type] = objectFormat(All)
+  implicit val byNameFormat: OFormat[ByName] = Json.format
+  implicit val byConfigFormat: OFormat[ByConfig] = Json.format
+  implicit val byFilesFormat: OFormat[ByFiles] = Json.format
+  implicit val orFormat: OFormat[Or] = getComplexFilterFormat(Or, Or.unapply)
+  implicit val andFormat: OFormat[And] = getComplexFilterFormat(And, And.unapply)
```

Comment:
[minor]
One last thing, can we please either mark them private or hide them inside
```scala
implicit val projectSearchFilterFormat: OFormat[ProjectSearchFilter] = {
  implicit val allFormat: OFormat[All.type] = objectFormat(All)
  // etc
}
```
This is required to hide these formatters and not let others to accidentally try to serialize / deserialize filters without type descriptor.

Reply:
Done